### PR TITLE
INTEG-465 Use Suggester component in Share document UI

### PR DESCRIPTION
**Pull request by trongtt**

### CODE REVIEW

File: apps/resources-wcm/src/main/webapp/javascript/eXo/wcm/frontoffice/public/ShareContent.js [hunk 1]
```diff
@@ -90,43 +90,32 @@
...
 
-    gj(".uiShareDocuments.resizable #textAreaInput").exoMentions({
-      onDataRequest : function(mode, query, callback) {
-        var url = window.location.protocol + '//' + window.location.host + '/' + eXo.env.portal.rest + '/social/people/getprofile/data.json?search=' + query;
-        gj.getJSON(url, function(responseData) {
-          responseData = _.filter(responseData, function(item) {
-            return item.name.toLowerCase().indexOf(query.toLowerCase()) > -1;
+    gj('.uiShareDocuments.resizable #textAreaInput').suggester({
+      type : "mix",
+      sourceProviders : ['exo:shareconent'],
```
Comment (thomasdelhomenie) [inline apps/resources-wcm/src/main/webapp/javascript/eXo/wcm/frontoffice/public/ShareContent.js]: Typo : shareconent -> sharecontent

Comment (trongtt) [reply apps/resources-wcm/src/main/webapp/javascript/eXo/wcm/frontoffice/public/ShareContent.js]: Fixed

**Comment by trongtt** [comment]
@thomasdelhomenie there is a technical limitation in suggester component that doesn't allow to do like that in this case. We have to improve suggester component to support initializing MIX mode from raw existing content (At the moment, I do not really have an idea to make it)

So I do think that the current fixed behavior is acceptable which is similar to the case of mentioning in Chat box.

WDYT ?

**Comment by thomasdelhomenie** [comment]
I don't think it is acceptable, it is a functional regression. I think we should do this improvement. Do we already have a jira issue for that ? Otherwise I guess we can do it as part of this issue, WDYT ?

> So I do think that the current fixed behavior is acceptable which is similar to the case of mentioning in Chat box.

Which box ?

**Comment by trongtt** [comment]
I mean the mentioning for task assignment plugin in the Chat input.

**Comment by thomasdelhomenie** [comment]
I guess you talk about the ++ syntax in the Chat message box (we have the blue background when we use the menu "Assign task"). Did we previously have this blue background here ? I don't remember. And it seems to not work on 4.4.x : http://plfentrial-4.4.x-snapshot-postgresql9.5.acceptance7.exoplatform.org

**Comment by trongtt** [comment]
++ syntax in Chat input is something newly supported. We didn't support the blue background here as it is a textarea.

**Comment by thomasdelhomenie** [comment]
ok, so there is no regression for the Chat, so that's not the same case than for Share Documents feature.

**Pull request closed without merging**